### ISSUE TITLE
use tuple-style imports to make diffs prettier and avoid long lines

### ISSUE THREAD

**Issue by chutz** [psf/black#1492]
**Describe the style change**:
It would be nice if it would always use this style as it makes diffs on import lines easier to read and shorter. It also makes long lines less likely in files that have a lot of imports.

**Examples in the current _Black_ style**:
Imports by default are single-line style. This style is much harder to read in diffs (sometimes it's quite difficult to determine what changed).
```
from foo import bar, baz, bla
from bar import foo
```

**Desired style**:

```
from foo import (
    bar,
    baz,
    bla,
)
from bar import (
    foo,
)
```

**Additional context**
Currently if it finds a tuple-style import that imports multiple names, it leaves it alone, but for ones that import a single name it collapses it. However it will not convert a line importing multiple names in to a tuple-style import making black inconsistent in this case.

**Comment by ichard26** [comment]
Hello @chutz!

Black already has a solution to your design change request. It's called the "magic trailing comma" where if a collection, function signature/call, or import statement has a trailing comma Black will explode it. Which means you can do this:
```py3
from typing import (List, Dict, Optional,)
```

and have Black format it to:
```py3
from typing import (
    List,
    Dict,
    Optional,
)
```

Without the trailing comma, Black will only explode an import statement if it violates the line-length limit. The problematic bug with the "magic trailing comma" feature regarding import statement is that Black collapses single name imports WITH a trailing comma. AFAIK Black only does this with imports... function signatures/calls, collections, etc. are untouched. 
```diff
(black) richard-26@ubuntu-laptop:~/programming/black$ black test.py --diff --color
--- test.py	2020-06-14 17:35:53.084183 +0000
+++ test.py	2020-06-14 17:35:59.146251 +0000
@@ -1,14 +1,14 @@
-from colorama import (Fore, Style,)
+from colorama import (
+    Fore,
+    Style,
+)
 from typing import (
     List,
     Dict,
 )
-from math import (
-    mean
-)
-from math import (
-    median,
-)
+from math import mean
+from math import median
+
 test = [
     "hello world",
 ]
would reformat test.py
All done! ✨ 🍰 ✨
1 file would be reformatted.
```

So IMO, this isn't really a design issue since it already has a workaround, but it does report a bug with the "magic trailing comma" regarding imports. I'd have expected the single name import with a trailing comma to stay exploded just like the `test` collection. This bug probably falls under the rather large issue of #1288. (And if that's the case, this would probably be considered as kinda a duplicate)

Thanks for the ~design issue~ bug report! Reducing diff churn is one of Black's goals :)

**Comment by joaoe** [comment]
While I understand the OP's request I personally prefer short import lines to remain collapsed. In large projects, expanding every import line would produce really long import sections. Regarding diffing, most tools support or can be configured to support substring diffs, e.g., github and gitlab do that for you, you can easily invoke git diff with word diffing and so on.

**Comment by ichard26** [comment]
Okay, I wasn't as clear enough with my response above. So I completely agree with you, short import lines should stay collapsed. By default, Black does this as long the import line doesn't violate the line-length limit. To the OP's request, I was providing a workaround/solution which is the "magic trailing comma". We're on the same page @joaoe :) I think providing some choice is a good idea, and the "magic trailing comma" helps with that.  
> So IMO, this isn't really a design issue since it already has a workaround, but it does report a bug with the "magic trailing comma" regarding imports.

**Comment by JelleZijlstra** [comment]
Closing this because the magic trailing comma covers this use case, as @ichard26 mentioned. I also just closed #1521 which asks us to enforce the opposite behavior.
